tso-server/src/main/java/com/yahoo/omid/tso/PersistenceProcessorImpl.java

```diff
@@ -27,57 +27,44 @@
 import javax.inject.Named;
 
 import org.jboss.netty.channel.Channel;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
 
-import com.google.common.annotations.VisibleForTesting;
 import com.google.common.util.concurrent.ThreadFactoryBuilder;
 import com.lmax.disruptor.BusySpinWaitStrategy;
 import com.lmax.disruptor.EventFactory;
 import com.lmax.disruptor.RingBuffer;
 import com.lmax.disruptor.SequenceBarrier;
-import com.lmax.disruptor.WorkHandler;
 import com.lmax.disruptor.WorkerPool;
 import com.yahoo.omid.committable.CommitTable;
-import com.yahoo.omid.metrics.Histogram;
 import com.yahoo.omid.metrics.Meter;
 import com.yahoo.omid.metrics.MetricsRegistry;
-import com.yahoo.omid.metrics.Timer;
+import com.yahoo.omid.tso.BatchPool.Batch;
 
-import com.yahoo.omid.tso.PersistenceProcessorImpl.PersistEvent.Type;
-import com.yahoo.omid.tso.PersistenceProcessorImpl.PersistenceProcessorHandler.Batch;
-import com.yahoo.omid.tso.PersistenceProcessorImpl.PersistenceProcessorHandler.BatchPool;
 
 class PersistenceProcessorImpl
     implements PersistenceProcessor {
 
     private static final long INITIAL_LWM_VALUE = -1L;
 
-    final ReplyProcessor reply;
-    final RetryProcessor retryProc;
-    final CommitTable.Client commitTableClient;
-    final CommitTable.Writer writer;
-    final Panicker panicker;
-    final RingBuffer<PersistBatchEvent> persistRing;
+    private final ReplyProcessor reply;
+    private final RingBuffer<PersistBatchEvent> persistRing;
 
-    private PersistenceProcessorHandler[] handlers;
-    private int numHandlers;
+    private final PersistenceProcessorHandler[] handlers;
+    private final int numHandlers;
 
-    private BatchPool batchPool;
-    private Batch batch;
-    private long batchIDCnt;
+    private final BatchPool batchPool;
+    public Batch batch;
 
+    volatile private long batchIDCnt;
     volatile private long lowWatermark = INITIAL_LWM_VALUE;
-    MonitoringContext lowWatermarkContext;
 
-    long lastFlush = System.nanoTime();
-    final static int BATCH_TIMEOUT_MS = 100;
+    MonitoringContext lowWatermarkContext;
 
     final Meter timeoutMeter;
 
     @Inject
     PersistenceProcessorImpl(TSOServerConfig config,
             MetricsRegistry metrics,
+            BatchPool batchPool,
```

Comment:
Resolved as you suggested. 
I added a provider that allocates an array and injects an array of handlers to PersistenceProcessorImpl.